src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -439,35 +446,31 @@ private void deleteSubElements(UUID elementUuid, String userId) {
      * @return ElementAttributes of element and all it's parents up to root directory
      */
     public List<ElementAttributes> getPath(UUID elementUuid, String userId) {
-        Optional<DirectoryElementEntity> currentElementOpt = repositoryService.getElementEntity(elementUuid);
-        ArrayList<ElementAttributes> path = new ArrayList<>();
-        boolean allowed;
-        if (currentElementOpt.isEmpty()) {
-            throw DirectoryException.createElementNotFound(ELEMENT, elementUuid);
-        }
-        DirectoryElementEntity currentElement = currentElementOpt.get();
+        DirectoryElementEntity currentElement = repositoryService.getElementEntity(elementUuid)
+                .orElseThrow(() -> DirectoryException.createElementNotFound(ELEMENT, elementUuid));
 
-        if (currentElement.getType().equals(DIRECTORY)) {
-            allowed = toElementAttributes(currentElement).isAllowed(userId);
-        } else {
-            allowed = toElementAttributes(repositoryService.getElementEntity(currentElement.getParentId()).orElseThrow()).isAllowed(userId);
-        }
+        boolean allowed = (Objects.equals(currentElement.getType(), DIRECTORY)) ?
+                toElementAttributes(currentElement).isAllowed(userId) :
+                toElementAttributes(getParentElement(currentElement)).isAllowed(userId);
 
         if (!allowed) {
             throw new DirectoryException(NOT_ALLOWED);
         }
 
+        List<ElementAttributes> path = new ArrayList<>();
         path.add(toElementAttributes(currentElement));
 
-        while (currentElement.getParentId() != null) {
-            currentElement = repositoryService.getElementEntity(currentElement.getParentId()).orElseThrow();
-            ElementAttributes currentElementAttributes = toElementAttributes(currentElement);
-            path.add(currentElementAttributes);
-        }
+        List<DirectoryElementEntity> ancestors = repositoryService.findAllAscendants(elementUuid, userId);
+        path.addAll(ancestors.stream().map(ElementAttributes::toElementAttributes).toList());
 
         return path;
     }
 
+    private DirectoryElementEntity getParentElement(@NotNull DirectoryElementEntity currentElement) {
+        return repositoryService.getElementEntity(currentElement.getParentId())
+                .orElseThrow(() -> DirectoryException.createElementNotFound(ELEMENT, currentElement.getParentId()));
+    }
+
```

Comment:
we already have a method: private ElementAttributes getParentElement(UUID elementUuid)